pkg/providers/amifamily/ami.go

```diff
@@ -263,7 +262,7 @@ func GetFilterAndOwnerSets(terms []v1beta1.AMISelectorTerm) (res []FiltersAndOwn
 				} else {
 					elem.Filters = append(elem.Filters, &ec2.Filter{
 						Name:   aws.String(fmt.Sprintf("tag:%s", k)),
-						Values: aws.StringSlice(functional.SplitCommaSeparatedString(v)),
```

Comment:
Wondering how this doesn't catch the bug: 
Looks like we should be trimming each value delimited by the commas. 
```
// SplitCommaSeparatedString splits a string by commas, removes whitespace, and returns
// a slice of strings
func SplitCommaSeparatedString(value string) []string {
	var result []string
	for _, value := range strings.Split(value, ",") {
		result = append(result, strings.TrimSpace(value))
	}
	return result
}
```